IMPORTING THE REQUIRED DEPENDENCIES

In [72]:
import numpy as np
import pandas as pd

LOADING THE DATASET

In [73]:
data = pd.read_csv('data\diabetesdata.csv')                             

VISUALISING THE DATASET

In [74]:
data

Unnamed: 0  Age Gender  BMI  Chol    TG   HDL   LDL    Cr   BUN  \
0              0   50      F   24  4.20  0.90  2.40  1.40  46.0  4.70   
1              1   26      M   23  3.70  1.40  1.10  2.10  62.0  4.50   
2              2   33      M   21  4.90  1.00  0.80  2.00  46.0  7.10   
3              3   45      F   21  2.90  1.00  1.00  1.50  24.0  2.30   
4              4   50      F   24  3.60  1.30  0.90  2.10  50.0  2.00   
...          ...  ...    ...  ...   ...   ...   ...   ...   ...   ...   
5127        5127   54      M   23  5.00  1.50  1.24  2.98  77.0  3.50   
5128        5128   50      F   22  4.37  2.09  1.37  2.29  47.3  4.40   
5129        5129   67      M   24  3.89  1.38  1.14  2.17  70.6  4.73   
5130        5130   60      F   29  5.91  1.29  1.73  2.85  50.2  7.33   
5131        5131   37      M   34  5.42  2.66  1.08  2.87  75.5  4.61   

      Diagnosis  
0             0  
1             0  
2             0  
3             0  
4             0  
...         ...  
5127          1  
5128          1  
5129          1  
5130          1  
5131          1  

[5132 rows x 11 columns]

In [75]:
data.head()

Unnamed: 0  Age Gender  BMI  Chol   TG  HDL  LDL    Cr  BUN  Diagnosis
0           0   50      F   24   4.2  0.9  2.4  1.4  46.0  4.7          0
1           1   26      M   23   3.7  1.4  1.1  2.1  62.0  4.5          0
2           2   33      M   21   4.9  1.0  0.8  2.0  46.0  7.1          0
3           3   45      F   21   2.9  1.0  1.0  1.5  24.0  2.3          0
4           4   50      F   24   3.6  1.3  0.9  2.1  50.0  2.0          0

WE ARE OBSERVING THE SHAPE OF THE DATASET AND OBSERVE THAT THE DATASET HAS 5132 ROWS AND 11 COLUMNS

In [76]:
data.shape

(5132, 11)

IN THE CELL BELOW WE OBSERVE THE DATATYPE OF EACH COLUMN IN THE DATASET AND THE NUMBER OF NON NULL ROWS IN EACH COLUMN

In [77]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5132 entries, 0 to 5131
Data columns (total 11 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  5132 non-null   int64  
 1   Age         5132 non-null   int64  
 2   Gender      5132 non-null   object 
 3   BMI         5132 non-null   int64  
 4   Chol        5132 non-null   float64
 5   TG          5132 non-null   float64
 6   HDL         5132 non-null   float64
 7   LDL         5132 non-null   float64
 8   Cr          5132 non-null   float64
 9   BUN         5132 non-null   float64
 10  Diagnosis   5132 non-null   int64  
dtypes: float64(6), int64(4), object(1)
memory usage: 441.2+ KB


WE ARE DROPPING THE UNNAMED:0 COLUNMN FROM THE PANDAS DATAFRAME AS IT IS IRRELEVANT TO OUR CODE AND CHANGING THE DATATYPE OF BMI FROM INT TO FLOAT AS IT IS MORE APPROPRIATE

In [78]:
data_new = data.drop(['Unnamed: 0'], axis=1)
data_new['BMI'] = data_new['BMI'].astype(float)

In [79]:
data_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5132 entries, 0 to 5131
Data columns (total 10 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        5132 non-null   int64  
 1   Gender     5132 non-null   object 
 2   BMI        5132 non-null   float64
 3   Chol       5132 non-null   float64
 4   TG         5132 non-null   float64
 5   HDL        5132 non-null   float64
 6   LDL        5132 non-null   float64
 7   Cr         5132 non-null   float64
 8   BUN        5132 non-null   float64
 9   Diagnosis  5132 non-null   int64  
dtypes: float64(7), int64(2), object(1)
memory usage: 401.1+ KB


WE ARE CALCULATING STATISTICAL FACTS LIKE THE MEAN , MEDIAN , STANDARS DEVIATION AND PERCENTILES FOR EACH APPLICABLE COLUMN IN THE DATAFRAME

In [80]:
data_new.Age.describe()

count    5132.000000
mean       48.950312
std        14.048794
min        20.000000
25%        36.000000
50%        49.000000
75%        59.000000
max        93.000000
Name: Age, dtype: float64

In [81]:
data_new.BMI.describe()

count    5132.000000
mean       24.613406
std         4.277205
min        15.000000
25%        22.000000
50%        24.000000
75%        27.000000
max        47.000000
Name: BMI, dtype: float64

In [82]:
data_new.Chol.describe()

count    5132.000000
mean        4.866882
std         1.001052
min         0.000000
25%         4.190000
50%         4.800000
75%         5.460000
max        11.650000
Name: Chol, dtype: float64

In [83]:
data_new.TG.describe()

count    5132.000000
mean        1.719328
std         1.327057
min         0.000000
25%         0.910000
50%         1.380000
75%         2.100000
max        32.640000
Name: TG, dtype: float64

In [84]:
data_new.HDL.describe()

count    5132.000000
mean        1.593305
std         1.038849
min         0.000000
25%         1.090000
50%         1.300000
75%         1.590000
max         9.900000
Name: HDL, dtype: float64

In [85]:
data_new.LDL.describe()

count    5132.000000
mean        2.914121
std         0.945423
min         0.300000
25%         2.290000
50%         2.790000
75%         3.400000
max         9.900000
Name: LDL, dtype: float64

In [86]:
data_new.Cr.describe()

count    5132.000000
mean       71.144800
std        28.494394
min         4.860753
25%        58.000000
50%        70.200000
75%        81.600000
max       800.000000
Name: Cr, dtype: float64

In [87]:
data_new.BUN.describe()

count    5132.000000
mean        4.896969
std         1.689206
min         0.500000
25%         3.900000
50%         4.720000
75%         5.600000
max        38.900000
Name: BUN, dtype: float64

WE ARE CALCULATING THE MEMORY OCCUPIED BY THE OLD AND NEW DATAFRAME IN BYTES AND WE CALCULATE ITS DIFFERENCE

In [88]:

memory_old = data.memory_usage(deep=True).sum()
memory_new = data_new.memory_usage(deep=True).sum()

memory_difference = memory_old - memory_new

print("Memory usage of old DataFrame:", memory_old)
print("Memory usage of new DataFrame:", memory_new)
print("Memory usage difference:", memory_difference)


Memory usage of old DataFrame: 708348
Memory usage of new DataFrame: 667292
Memory usage difference: 41056
